R/DiseasyModel.R

```diff
@@ -0,0 +1,200 @@
+#' @title Base module for diseasy model templates
+#'
+#' @description
+#'   The `DiseasyModel` module implements common functionality that all models have available beyond that provided by
+#'   `DiseasyBaseModule`.
+#'   Most notably, the model module facilitates:
+#'   * Module interfaces:
+#'     The module contains the functional modules via its active bindings:
+#'     * `$activity`: `DiseasyActivity`
+#'     * `$observables`: `DiseasyObservables`
+#'     * `$season`: `DiseasySeason`
+#'     Configured instances of these modules can be provided during initialisation. If not, default instances are
+#'     created.
+#'   * Model interface:
+#'     The module defines the functions `$get_results()`, `$get_training_data()` and the `$parameters` binding.
+#'     These functions define the "API" of the models and ensure the models can take part in the ensemble.
+#' @examples
+#'   # Normally, you would not want to create this module directly, but it is possible.
+#'   Model_module <- DiseasyModel$new()
+#'
+#'   rm(Model_module)
+#' @return
+#'   A new instance of the `DiseasyModel` [R6][R6::R6Class] class.
+#' @export
+#' @seealso [lgr][lgr::lgr]
+DiseasyModel <- R6::R6Class(                                                                                            # nolint: object_name_linter
+  classname = "DiseasyModel",
+  inherit = DiseasyBaseModule,
+
+  public = list(
+
+    #' @description
+    #'   Creates a new instance of the `DiseasyModel` [R6][R6::R6Class] class.
+    #'   This module is typically not constructed directly but rather through `DiseasyModel*` classes
+    #' @param activity,observables,season (`boolean` or `R6::R6Class instance`)\cr
```

Comment:
Is it on purpose that there are no spaces in "activity,observables,season"?

Reply:
Yes, this way `Roxygen` renders it nicely.

![billede](https://github.com/user-attachments/assets/ba89a891-624f-4813-b641-a5bf7a078bd5)